src/screens/RegisterENS/RegisterENS.js

```diff
@@ -0,0 +1,295 @@
+  const estimationErrorMessage = useRootSelector((root) => root.transactionEstimate.errorMessage);
+
+  const [usernameValue, setUsernameValue] = useState(!ensName?.username ? null : ensName?.username);
+  const [isCheckingUsername, setIsCheckingUsername] = useState(false);
+
+  const gasAddress = getGasAddress(CHAIN.ETHEREUM, feeInfo?.gasToken);
+  const isUsernameInputDirty = usernameValue !== null;
+  const usernameValidationErrorMessage = isUsernameInputDirty ? validateUsername(usernameValue) : null;
+  const buttonDisable = !!usernameValidationErrorMessage || !!errorMessage || isCheckingUsername;
+  const isEditable = ensName?.username;
+  const showFeeValue = !estimationErrorMessage && !!user;
+  const showENSSaveButton = !!usernameValue && !feeInfo;
+
+  const getButtonName = () => {
+    if (!!usernameValidationErrorMessage && !errorMessage) return usernameValidationErrorMessage;
+    if (!usernameValidationErrorMessage && errorMessage) return errorMessage;
+    return t('button.save');
+  };
+
+  const reserveENSName = () => {
+    if (user?.username) dispatch(claimENSNameAction(user?.username));
+  };
+
+  const sendENSTransaction = () => {
+    // const statusCallback = (transactionStatus: TransactionStatus) => {
+    //   // console.log('statusCallback', statusCallback)
+    //   navigation.dismiss();
+    //   navigation.navigate(SEND_TOKEN_TRANSACTION, {
+    //     ...transactionStatus,
+    //     noRetry: true,
+    //     goBackDismiss: true,
+    //   });
+    // };
+    // dispatch(sendENSTransaction(statusCallback));
+  };
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const onValidUsername = useCallback(
+    debounce(() => {
+      if (usernameValue) dispatch(checkUsernameAvailabilityAction(usernameValue));
+    }, 200),
+    [usernameValue],
+  );
+
+  useEffect(() => {
+    // prepare for username check if no user set
+    if (!user) dispatch(resetUsernameCheckAction(true));
+    if (ensName?.username) dispatch(claimENSNameAction(ensName?.username));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (!isCheckingUsername && !usernameValidationErrorMessage && !errorMessage && isUsernameInputDirty) {
+      setIsCheckingUsername(true);
+    } else if (isCheckingUsername && (usernameValidationErrorMessage || errorMessage)) {
+      // reset if error occurred during update
+      setIsCheckingUsername(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [usernameValue, errorMessage]);
+
+  useEffect(() => {
+    // user updated, reset
+    if (isCheckingUsername) setIsCheckingUsername(false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [user]);
+
+  useEffect(() => {
+    if (!usernameValidationErrorMessage) onValidUsername();
+    return onValidUsername.cancel;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [onValidUsername, usernameValue]);
```

Comment:
Do we need all the separate `useEffect`s? Just wondering if any can be merged together, unless is there a reason for the separation?

Reply:
@IAmKio The dependency array is changed in every useEffect.